handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -614,26 +618,45 @@ public final SSLEngineResult wrap(
                 }
             }
 
-            if (dst.remaining() < MAX_ENCRYPTED_PACKET_LENGTH) {
+            int endOffset = offset + length;
+            int srcsLen = 0;
+
+            for (int i = offset; i < endOffset; ++i) {
+                final ByteBuffer src = srcs[i];
+                if (src == null) {
+                    throw new IllegalArgumentException("srcs[" + i + "] is null");
+                }
+                if (srcsLen == MAX_PLAINTEXT_LENGTH) {
+                    continue;
+                }
+
+                srcsLen += src.remaining();
+                if (srcsLen > MAX_PLAINTEXT_LENGTH) {
```

Comment:
immagine sizes of srcs are: {2, `Integer.MAX_VALUE`}. Then we will overflow `srcsLen`.